gammapy/estimators/flux_map.py

```diff
@@ -0,0 +1,399 @@
+        the maps dictionary. Expected entries are the following:
+        * norm : the norm factor
+        * norm_err : optional, the error on the norm factor.
+        * norm_errn : optional, the negative error on the norm factor.
+        * norm_errp : optional, the positive error on the norm factor.
+        * norm_ul : optional, the upper limit on the norm factor.
+        * norm_scan : optional, the norm values of the test statistic scan.
+        * stat_scan : optional, the test statistic scan values.
+        * ts : optional, the delta TS associated with the flux value.
+        * counts : optional, the number counts value.
+    reference_model : `~gammapy.modeling.models.SkyModel`, optional
+        the reference model to use for conversions. Default in None.
+        If None, a model consisting of a point source with a power law spectrum of index 2 is assumed.
+    gti : `~gammapy.data.GTI`
+        the maps GTI information. Default is None.
+    """
+    def __init__(self, data, reference_model=None, gti=None):
+        self.geom = data['norm'].geom
+
+        if reference_model == None:
+            log.warning("No reference model set for FluxMap. Assuming point source with E^-2 spectrum.")
+            reference_model = self._default_model()
+
+        self.reference_model = reference_model
+
+        self.gti = gti
+
+        super().__init__(data, spectral_model=reference_model.spectral_model)
+
+    @staticmethod
+    def _default_model():
+        return SkyModel(spatial_model=PointSpatialModel(), spectral_model=PowerLawSpectralModel(index=2))
+
+    @property
+    def e_ref(self):
+        axis = self.geom.axes["energy"]
+        return axis.center
+
+    @property
+    def e_min(self):
+        axis = self.geom.axes["energy"]
+        return axis.edges[:-1]
+
+    @property
+    def e_max(self):
+        axis = self.geom.axes["energy"]
+        return axis.edges[1:]
+
+    def get_flux_points(self, coord=None):
+        """Extract flux point at a given position.
+
+        The flux points are returned in the the form of a `~gammapy.estimators.FluxPoints` object
+        (which stores the flux points in an `~astropy.table.Table`)
+
+        Parameters
+        ---------
+        coord : `~astropy.coordinates.SkyCoord`
+            the coordinate where the flux points are extracted.
+
+        Returns
+        -------
+        fluxpoints : `~gammapy.estimators.FluxPoints`
+            the flux points object
+        """
+        if coord is None:
+            coord = self.geom.center_skydir
+        energies = self.e_ref
+        coords = MapCoord.create(dict(skycoord=coord, energy=energies))
+
+        ref = self.dnde_ref
+
+        fp = dict()
+        fp["norm"] = self.norm.get_by_coord(coords) * self.norm.unit
+
+        for quantity in self._available_quantities:
+            norm_quantity = f"norm_{quantity}"
+            res = self.__getattribute__(norm_quantity).get_by_coord(coords)
```

Comment:
I think you can directly use `geattr(self, norm_quantity)` to avoid the access to the internal method...

Reply:
Done